react/redux-01/src/features/accounts/accountSlice.js

```diff
@@ -0,0 +1,149 @@
+// we can write mutate code within createSlice, and package 'immer' will
+// convert it to immuate code for us behind the scene.
+
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  balance: 0,
+  loan: 0,
+  loanPurpose: '',
+  isLoading: false,
+};
+
+const accountSlice = createSlice({
+  name: 'account',
+  initialState,
+  reducers: {
+    deposit(state, action) {
+      state.balance += action.payload;
+      state.isLoading = false;
+    },
+    withdraw(state, action) {
+      state.balance -= action.payload;
+    },
+    requestLoan: {
+      // the actions created by RTK can only accepted one argument
+      // if you want to accept more than one argument, you have to turn this
+      // action into object, and setup a `prepare()` method to define a payload object
+      prepare(amount, purpose) {
```

Comment:
prepare method is needed if there are more than one payload, or there are some side effects need to be handled.